probe-rs/src/vendor/nxp/sequences/nxp_armv8m.rs

```diff
@@ -725,13 +726,83 @@ impl MIMXRT118x {
         Self(())
     }
 
-    /// Create a sequence handle for the MIMXRT117x.
+    /// Create a sequence handle for the MIMXRT118x.
     pub fn create() -> Arc<dyn ArmDebugSequence> {
         Arc::new(Self::new())
     }
+
+    /// Necessary for AIRCR.SYSRESETREQ to not be a noop
+    /// Reference:
+    /// binaries/Scripts/RT1180_connect_M33_wake_M7.scp:53 in Linkserver
+    /// `27.3.3.2` in `i.MX RT1180 Reference Manual, Rev. 6, 09/2024`
+    fn clear_cm33_reset_mask(
+        &self,
+        interface: &mut dyn ArmMemoryInterface,
+    ) -> Result<(), ArmError> {
+        const SRC_SRMASK: u64 = 0x54460018;
+        let mut srmask = interface.read_word_32(SRC_SRMASK)?;
+        tracing::trace!("SRC.SRMASK: {srmask:#010X}. Clearing the CM33_RESET_MASK mask...");
+        srmask &= !(0b1 << 8);
```

Comment:
Would it be possible to make a proper [bytefield](https://docs.rs/bitfield/latest/bitfield/) based typed reg impl instead of magic constants? :)

Reply:
smh, complaining about the state of the art register masks. How could a fellow embedded engineer do this to me 🤌🤌 Gotta fix it soon 🔥

Reply:
Hey, I write x86 code :D